modules/azure/api_management_api/main.tf

```diff
@@ -41,7 +41,7 @@ resource "azurerm_api_management_api" "api" {
     authorization_server_name = "${lower(replace(var.api_settings.name, " ", "-"))}-auth"
   }
 
-  soap_pass_through = var.soap_pass_through
+  api_type = var.api_type == null && var.soap_pass_through ? "soap" : var.api_type != null ? var.api_type : "http"
```

Comment:
since it is a ternary operator I think the first condition should be wrapped in brackets

(var.api_type == null && var.soap_pass_through) ? "soap" : var.api_type != null ? var.api_type : "http"

Reply:
Changed as suggested